### PR TITLE
[ECO-4947] Spec complete for typing indicators 

### PR DESCRIPTION
**Pull request by umair-ably**
Spec complete for typing indicators 

https://github.com/user-attachments/assets/36f08ea1-c91e-4fd2-9030-5bf5d67a8d5c

add unit tests as part of [this](https://github.com/ably-labs/ably-chat-swift/issues/140) issue

## Release Notes

- **New Features**
  - Enhanced chat functionality with typing notifications.
  - Introduced a dedicated class for managing typing indicators.
  - Improved connection status monitoring and error handling.

- **Bug Fixes**
  - Resolved issues with typing indicator delays and connection status updates.

- **Documentation**
  - Updated comments and documentation for clarity on typing features and options.

- **Tests**
  - Expanded integration tests to cover typing indicators and their behaviors.

### CODE REVIEW

**Comment by umair-ably** [comment]
@coderabbitai pause

File: Sources/AblyChat/TimerManager.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+internal final actor TimerManager {
```
Comment (maratal) [inline Sources/AblyChat/TimerManager.swift:3]: I've already seen this actor in the previous PR.

Comment (maratal) [reply Sources/AblyChat/TimerManager.swift:3]: Ok, you moved it from elsewhere.

**Review by maratal — APPROVED** [review]
LGTM

**Comment by umair-ably** [comment]
> I would suggest filter out own typings.

some users might not run it on 2 separate simulators so i think it's still nice to see in the demo app

**Pull request merged**